editor/utils/mobile/index.js

```diff
@@ -18,20 +7,26 @@ const isMobileChecker = () => 'object' === typeof window && window.innerWidth <
  *
  * @return {Object}            rehydrate payload with isSidebarOpened disabled if on mobile
  */
-export const disableIsSidebarOpenedOnMobile = ( payload, isMobile = isMobileChecker() ) => (
-	isMobile ? { ...payload, isSidebarOpened: false } : payload
+export const disableIsSidebarOpenedOnMobile = ( payload ) => (
+	payload.isSidebarOpenedMobile ? { ...payload, isSidebarOpenedMobile: false } : payload
 );
 
 /**
  * Middleware
  */
 
-export const mobileMiddleware = () => next => action => {
+export const mobileMiddleware = ( { getState } ) => next => action => {
 	if ( action.type === 'REDUX_REHYDRATE' ) {
 		return next( {
 			type: 'REDUX_REHYDRATE',
 			payload: disableIsSidebarOpenedOnMobile( action.payload ),
 		} );
 	}
+	if ( action.type === 'TOGGLE_SIDEBAR' ) {
+		return next( {
```

Comment:
You can code it using `toggleSidebar( isMobile( getState() )` to take advantage of the existing `toggleSidebar` action creator.